From: Guoqing Jiang <guoqing.jiang@cloud.ionos.com>
Date: Thu, 9 Jan 2020 14:40:43 +0100
Subject: RDMA/core: Remove err in iw_query_port
Patch-mainline: v5.6-rc1
Git-commit: 1e123d96b85fd41278efc994f8096e1df86fd427
References: jsc#SLE-8449

Since we can return device->ops.query_port directly, so no need to keep
those lines.

Link: https://lore.kernel.org/r/20200109134043.15568-1-guoqing.jiang@cloud.ionos.com
Signed-off-by: Guoqing Jiang <guoqing.jiang@cloud.ionos.com>
Signed-off-by: Jason Gunthorpe <jgg@mellanox.com>
Acked-by: Thomas Bogendoerfer <tbogendoerfer@suse.de>
---
 drivers/infiniband/core/device.c |    7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

--- a/drivers/infiniband/core/device.c
+++ b/drivers/infiniband/core/device.c
@@ -1977,7 +1977,6 @@ static int iw_query_port(struct ib_devic
 {
 	struct in_device *inetdev;
 	struct net_device *netdev;
-	int err;
 
 	memset(port_attr, 0, sizeof(*port_attr));
 
@@ -2008,11 +2007,7 @@ static int iw_query_port(struct ib_devic
 	}
 
 	dev_put(netdev);
-	err = device->ops.query_port(device, port_num, port_attr);
-	if (err)
-		return err;
-
-	return 0;
+	return device->ops.query_port(device, port_num, port_attr);
 }
 
 static int __ib_query_port(struct ib_device *device,
